packages/indexer/src/sinks/vcr.ts

```diff
@@ -0,0 +1,16 @@
+import { Sink, type SinkWriteArgs } from "../sink";
+import type { VcrReplayResult } from "../vcr";
+
+export class VcrSink<TData> extends Sink<TData> {
+  public result: VcrReplayResult<TData>["outputs"] = [];
+
+  async write({ data, endCursor }: SinkWriteArgs<TData>) {
+    this.emit("write", { data });
+    this.result.push({ data, endCursor });
+    this.emit("flush");
+  }
+}
+
+export function vcr<TData>() {
+  return new VcrSink<TData>();
```

Comment:
```suggestion
export function vcr<TData extends Record<string, unknown>>() {
```

Reply:
In my case if we don't add this I still get some type errors because `Tdata` doesn't satisfy the constraint `Record<string, unknown>`.